chore(build): add devcontainer

DanielHabenicht opened the pull request:
I am using mostly `devcontainers` for local development nowadays so I just wanted to give you the option to add it to your repository too.
If you don't want or use it it just close the PR. :)

This would allow anybody to jumpstart developing on this repository by cloning it with vscode (and docker) or simply by using github codespaces.

<!--- Please keep this note for the community --->
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
<!--- Thank you for keeping this note for the community --->


<!--- Release note for [CHANGELOG](https://github.com/bpg/terraform-provider-proxmox/blob/main/CHANGELOG.md) will be created automatically using the PR's title, update it accordingly. --->

bpg commented:
Works nicely, thanks! We should keep it :)

Just one comment, i had to update the base image to `mcr.microsoft.com/devcontainers/go`, the "universal" one doesn't seem to be available for for all platforms:
<img width="800" alt="Screenshot 2023-11-09 at 9 51 49 PM" src="https://github.com/bpg/terraform-provider-proxmox/assets/627562/b292fb53-2448-4ad0-b4bf-05f11e40d272">

DanielHabenicht commented:
Updated the base image and sign off commits. 
I also added it to the devcontainer configuration as it seems to be a github bug - I hope you don't mind if Github also signed off on it (just the first two commits) :D

bpg reviewed (approved):
Thank you @DanielHabenicht!

LGTM! 🚀
